drivers/adc/adc_nrfx_saadc.c

```diff
@@ -384,38 +415,44 @@ static int adc_nrfx_read_async(const struct device *dev,
 
 static void saadc_irq_handler(const struct device *dev)
 {
-	if (nrf_saadc_event_check(NRF_SAADC, NRF_SAADC_EVENT_END)) {
-		nrf_saadc_event_clear(NRF_SAADC, NRF_SAADC_EVENT_END);
+	nrfx_saadc_evt_type_t event;
+	nrfx_err_t err;
 
-		nrf_saadc_task_trigger(NRF_SAADC, NRF_SAADC_TASK_STOP);
-		nrf_saadc_disable(NRF_SAADC);
+	event = m_data.event->type;
 
+	if (event == NRFX_SAADC_EVT_DONE) {
 		adc_context_on_sampling_done(&m_data.ctx, dev);
-	} else if (nrf_saadc_event_check(NRF_SAADC,
-					 NRF_SAADC_EVENT_CALIBRATEDONE)) {
-		nrf_saadc_event_clear(NRF_SAADC, NRF_SAADC_EVENT_CALIBRATEDONE);
-
-		/*
-		 * The workaround for Nordic nRF52832 anomalies 86 and
-		 * 178 is an explicit STOP after CALIBRATEOFFSET
-		 * before issuing START.
-		 */
-		nrf_saadc_task_trigger(NRF_SAADC, NRF_SAADC_TASK_STOP);
-		nrf_saadc_task_trigger(NRF_SAADC, NRF_SAADC_TASK_START);
-		nrf_saadc_task_trigger(NRF_SAADC, NRF_SAADC_TASK_SAMPLE);
+	} else if (event == NRFX_SAADC_EVT_CALIBRATEDONE) {
+		err = nrfx_saadc_mode_trigger();
+		if (err != NRFX_SUCCESS) {
+			LOG_ERR("Cannot start sampling: %d", err);
+		}
+
 	}
 }
 
+static void irq_nrfx_event_handler(nrfx_saadc_evt_t const *event)
```

Comment:
Just for consistency:
```suggestion
static void irq_nrfx_event_handler(const nrfx_saadc_evt_t *event)
```